### PR TITLE
feature 1586 added wrapper-specific version of probabilistic config variables

### LINKED ISSUE #1586: Allow FCST_IS_PROB variable setting specific to tool (FCST_<tool_name>_IS_PROB)
## Describe the Enhancement ##
Currently, setting FCST_IS_PROB to True in one tool will cause any additional tool being called in a use case to also have FCST_IS_PROB = True, no matter where in the process list that variable is set. This is problematic, for instance in use cases where uncalibrated probabilistic data is being created from GenEnsProd output, but tools prior to GenEnsProd are reading in continuous data fields.

The current system has a workaround using user-defined instances for the probabilistic tool. For example, if 3 tools are called in the following order:

`PROCESS_LIST = SeriesAnalysis, GenEnsProd, GridStat`

and GridStat needs FCST_IS_PROB = True while the other two tools don't, they can use:

`PROCESS_LIST = SeriesAnalysis, GenEnsProd, GridStat(my_instance)`

and 

```
[my_instance]
FCST_IS_PROB = True
```

will produce the desired output. The successful resolution of this issue will mean that the workaround is no longer needed and users can use

`FCST_GRID_STAT_IS_PROB = True`

instead.

### Time Estimate ###
Engineer to provide better estimate

### Sub-Issues ###
Consider breaking the enhancement down into sub-issues.
- [ ] *Add a checkbox for each sub-issue here.*

### Relevant Deadlines ###
*List relevant project deadlines here or state NONE.*

### Funding Source ###
*Define the source of funding and account keys here or state NONE.*

### Assignee ###
- [ ] Select **engineer(s)** or **no engineer** required
- [ ] Select **scientist(s)** or **no scientist** required

### Labels ###
- [ ] Select **component(s)**
- [ ] Select **priority**
- [ ] Select **requestor(s)**

### Projects and Milestone ###
- [ ] Select **Repository** and/or **Organization** level **Project(s)** or add **alert: NEED PROJECT ASSIGNMENT** label
- [ ] Select **Milestone** as the next official version or **Future Versions**

## Define Related Issue(s) ##
Consider the impact to the other METplus components.
- [ ] [METplus](https://github.com/dtcenter/METplus/issues/new/choose), [MET](https://github.com/dtcenter/MET/issues/new/choose), [METdatadb](https://github.com/dtcenter/METdatadb/issues/new/choose), [METviewer](https://github.com/dtcenter/METviewer/issues/new/choose), [METexpress](https://github.com/dtcenter/METexpress/issues/new/choose), [METcalcpy](https://github.com/dtcenter/METcalcpy/issues/new/choose), [METplotpy](https://github.com/dtcenter/METplotpy/issues/new/choose)

## Enhancement Checklist ##
See the [METplus Workflow](https://metplus.readthedocs.io/en/latest/Contributors_Guide/github_workflow.html) for details.
- [ ] Complete the issue definition above, including the **Time Estimate** and **Funding Source**.
- [ ] Fork this repository or create a branch of **develop**.
Branch name: `feature_<Issue Number>_<Description>`
- [ ] Complete the development and test your changes.
- [ ] Add/update log messages for easier debugging.
- [ ] Add/update unit tests.
- [ ] Add/update documentation.
- [ ] Add any new Python packages to the [METplus Components Python Requirements](https://metplus.readthedocs.io/en/develop/Users_Guide/overview.html#metplus-components-python-requirements) table.
- [ ] Push local changes to GitHub.
- [ ] Submit a pull request to merge into **develop**.
Pull request: `feature <Issue Number> <Description>`
- [ ] Define the pull request metadata, as permissions allow.
Select: **Reviewer(s)** and **Linked issues**
Select: **Repository** level development cycle **Project** for the next official release
Select: **Milestone** as the next official version
- [ ] Iterate until the reviewer(s) accept and merge your changes.
- [ ] Delete your fork or branch.
- [ ] Close this issue.

### PR DESCRIPTION
**Pull request by georgemccabe**
## Pull Request Testing ##

- [X] Describe testing already performed for these changes:</br>

Added unit tests to ensure correct values are read from config variables

- [X] Recommend testing for the reviewer(s) to perform, including the location of input datasets, and any additional instructions:</br>

Test that your use case that requires this change will now work properly.

- [X] Do these changes include sufficient documentation updates, ensuring that no errors or warnings exist in the build of the documentation? **[Yes]**

- [X] Do these changes include sufficient testing updates? **[Yes]**

- [X] Will this PR result in changes to the test suite? **[No]**</br>

- [X] Please complete this pull request review by **6/10/2022**.</br>

## Pull Request Checklist ##
See the [METplus Workflow](https://metplus.readthedocs.io/en/latest/Contributors_Guide/github_workflow.html) for details.
- [X] Add any new Python packages to the [METplus Components Python Requirements](https://metplus.readthedocs.io/en/develop/Users_Guide/overview.html#metplus-components-python-requirements) table.
- [X] Review the source issue metadata (required labels, projects, and milestone).
- [X] Complete the PR definition above.
- [X] Ensure the PR title matches the feature or bugfix branch name.
- [X] Define the PR metadata, as permissions allow.
Select: **Reviewer(s)**
Select: **Organization** level software support **Project** or **Repository** level development cycle **Project**
Select: **Milestone** as the version that will include these changes
- [x] After submitting the PR, select **Linked issues** with the original issue number.
- [ ] After the PR is approved, merge your changes. If permissions do not allow this, request that the reviewer do the merge.
- [ ] Close the linked issue and delete your feature or bugfix branch from GitHub.
…not need to be a public method

### CODE REVIEW

**Comment by georgemccabe** [comment]
Note that an expired SSL certificate is causing the 2 failing "Update Data Volumes" checks. This PR does not involve adding new input data so these checks are not needed to pass for the purposes of the review.

File: internal_tests/pytests/grid_stat/test_grid_stat_wrapper.py [hunk 1]
```diff
@@ -51,6 +51,46 @@ def set_minimum_config_settings(config):
     config.set('config', 'OBS_VAR1_NAME', obs_name)
     config.set('config', 'OBS_VAR1_LEVELS', obs_level)
 
+@pytest.mark.parametrize(
```
Comment (j-opatz) [inline internal_tests/pytests/grid_stat/test_grid_stat_wrapper.py:54]: This logic is interesting and caught my eye. I'm assuming it's critical to how the various scenarios are evaluated of fcst/obs probablity and the various tools now having their own settings. 
Can you summarize the logic of this for me? For example, if "FCST_IS_PROB" is False but "FCST_GRID_STAT_IS_PROB" is True, how is that handled? And perhaps more importantly, how is the reverse handled: where FCST_GRID_STAT_IS_PROB is False but FCST_IS_PROB is True? What about the scenario where GridStat is the only tool called but these conditions are present?

**Comment by j-opatz** [comment]
Once the comment above is addressed, this PR will be approved. I already tested this against the use case this originated from and am satisfied the fix addresses the issue.

File: internal_tests/pytests/grid_stat/test_grid_stat_wrapper.py (hunk 1 above)
Comment (georgemccabe) [reply internal_tests/pytests/grid_stat/test_grid_stat_wrapper.py:54]: These are the different values that are passed to the unit test. In this test, there is a list of tuples that contain config overrides and expected values. For each tuple in the list, the values are passed to the test function and used in that call to the test. The values in config_overrides are set in the METplusConfig before initializing the wrapper. The values in expected_values are compared to the values set in the wrapper's c_dict dictionary to make sure they are set as expected.

The expected logic is:
* If the wrapper specific version (FCST_GRID_STAT_IS_PROB) is set, use that value. Otherwise:
* If the generic version (FCST_IS_PROB) is set, use that value. Otherwise:
* Set the variable to False

The first example you provided ("FCST_IS_PROB" is False but "FCST_GRID_STAT_IS_PROB" is True) is not explicitly tested here, so it wouldn't hurt to add it. I can do that. Test 5 ensures that the wrapper-specific version is read properly when the generic is unset ("FCST_IS_PROB" is unset but "FCST_GRID_STAT_IS_PROB" is True).

Test 3 covers your 2nd example: FCST_GRID_STAT_IS_PROB is False but FCST_IS_PROB is True

If GridStat is the only tool called, then the same rules listed above apply.

Comment (j-opatz) [reply internal_tests/pytests/grid_stat/test_grid_stat_wrapper.py:54]: Excellent, thank you for the explanation and additional test.

**Review by j-opatz — APPROVED** [review]
With the comment on competing Prob settings resolved, this PR is approved.
